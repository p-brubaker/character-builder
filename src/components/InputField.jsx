import React from 'react'

function InputField() {
  return <div>Input field goes here</div>
}

export default InputField
