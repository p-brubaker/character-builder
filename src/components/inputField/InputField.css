.input-field {
  width: 100%;
  display: flex;
  justify-content: center;
  flex-direction: column;
  align-items: center;
  overflow: scroll;
}

.picker {
  display: flex;
  flex-direction: column;
}
