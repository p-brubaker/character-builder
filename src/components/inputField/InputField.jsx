import React from 'react'
import './InputField.css'

function InputField(props) {
  const {
    head,
    middle,
    bottom,
    catchphrasesInput,
    handleSelect,
    setCatchphrasesInput,
    updateCatchphrases,
  } = props

  return (
    <div className="input-field">
      <div className="picker">
        <label htmlFor="Head">
          Head
          <select value={head} name="Head" onChange={(e) => handleSelect(e.target.value, 'Head')}>
            {['Duck', 'Dog', 'Bird', 'Horse'].map((choice) => (
              <option value={choice} key={choice}>
                {choice}
              </option>
            ))}
          </select>
        </label>
        <label htmlFor="Middle">
          Middle
          <select
            value={middle}
            name="Middle"
            onChange={(e) => handleSelect(e.target.value, 'Middle')}
          >
            {['Fancy', 'Pink', 'Red', 'Blue'].map((choice) => (
              <option value={choice} key={choice}>
                {choice}
              </option>
            ))}
          </select>
        </label>
        <label htmlFor="Bottom">
          Bottom
          <select
            value={bottom}
            name="Bottom"
            onChange={(e) => handleSelect(e.target.value, 'Bottom')}
          >
            {['White', 'Blue', 'Leg'].map((choice) => (
              <option value={choice} key={choice}>
                {choice}
              </option>
            ))}
          </select>
        </label>
      </div>
      <div>
        <label htmlFor="catchphrases-input">Add a catch phrase</label>
        <input
          type="text"
          value={catchphrasesInput}
          onChange={(e) => setCatchphrasesInput(e.target.value)}
          name="catchphrases-input"
          id="catchphrases-input"
        />
        <button type="submit" onClick={() => updateCatchphrases(catchphrasesInput)}>
          Add
        </button>
      </div>
    </div>
  )
}

export default InputField
