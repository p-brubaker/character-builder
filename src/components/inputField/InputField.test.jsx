import { render } from '@testing-library/react'
import InputField from '../inputField/InputField'

it('should render the InputField component', () => {
  const { container } = render(<InputField head="Duck" middle="Blue" bottom={'blue'} />)
  expect(container).toMatchSnapshot()
})
