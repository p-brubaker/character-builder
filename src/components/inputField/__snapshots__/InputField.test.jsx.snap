// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the InputField component 1`] = `
<div>
  <div
    class="input-field"
  >
    <div
      class="picker"
    >
      <label
        for="Head"
      >
        Head
        <select
          name="Head"
        >
          <option
            value="Duck"
          >
            Duck
          </option>
          <option
            value="Dog"
          >
            Dog
          </option>
          <option
            value="Bird"
          >
            Bird
          </option>
          <option
            value="Horse"
          >
            Horse
          </option>
        </select>
      </label>
      <label
        for="Middle"
      >
        Middle
        <select
          name="Middle"
        >
          <option
            value="Fancy"
          >
            Fancy
          </option>
          <option
            value="Pink"
          >
            Pink
          </option>
          <option
            value="Red"
          >
            Red
          </option>
          <option
            value="Blue"
          >
            Blue
          </option>
        </select>
      </label>
      <label
        for="Bottom"
      >
        Bottom
        <select
          name="Bottom"
        >
          <option
            value="White"
          >
            White
          </option>
          <option
            value="Blue"
          >
            Blue
          </option>
          <option
            value="Leg"
          >
            Leg
          </option>
        </select>
      </label>
    </div>
    <div>
      <label
        for="catchphrases-input"
      >
        Add a catch phrase
      </label>
      <input
        id="catchphrases-input"
        name="catchphrases-input"
        type="text"
        value=""
      />
      <button
        type="submit"
      >
        Add
      </button>
    </div>
  </div>
</div>
`;
